### DISCUSSION TITLE
mxnet doesn't recognize GPU on WSL2

### DISCUSSION THREAD

**Discussion by Nakroma** [apache/incubator-mxnet#20672] [Q&A]
Hello, I'm trying to run mxnet on WSL2 with my GeForce 3060 Ti and CUDA 11.2, however `mx.util.get_gpu_count()` returns 0, and all operations that have to do with GPU return `MXNetError: Check failed: device_count_ > 0 (-1 vs. 0) : GPU usage requires at least 1 GPU`.

nvcc --version:
```
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2021 NVIDIA Corporation
Built on Sun_Feb_14_21:12:58_PST_2021
Cuda compilation tools, release 11.2, V11.2.152
Build cuda_11.2.r11.2/compiler.29618528_0
```

pip lists `mxnet-cu112==1.8.0.post0` as installed, I've also installed the WSL driver from here https://developer.nvidia.com/cuda/wsl  
I'm a bit lost on what to do here at this point

**Comment by szha** [comment]
@TristonC any suggestions on the WSL support?

**Comment by dai-ichiro** [comment]
My environment is 

Ubuntu-20.04LTS on WSL2
python 3.8.10
mxnet-cu112==1.8.0.post0

And MXNet can detect GPU.

How is nvidia-smi outputs? 
My outputs is as below
![スクリーンショット 2021-10-19 212349](https://user-images.githubusercontent.com/53703225/137908451-da3d0062-5493-4713-9b9a-d14b90102c0a.png)

**Comment by Nakroma** [comment]
I get this when running it
```
Failed to initialize NVML: GPU access blocked by the operating system
Failed to properly shut down NVML: GPU access blocked by the operating system
```
thought this was because NVML isnt supported on WSL2?

**Comment by dai-ichiro** [comment]
What is your Windows Build number?
My Windows Build is 22000, and I am using mxnet-cu112==1.8.0.post0 with no trouble.

This site may help you.
https://forums.developer.nvidia.com/t/failed-to-initialize-nvml-gpu-access-blocked-by-the-operating-system/174610

**Comment by Nakroma** [comment]
I'm on 19043.1288. I wanted to update to the windows preview build, but https://insider.windows.com/ is down for me for days now. Could that be it?
